lib/improver/argparser.py

```diff
@@ -28,19 +28,185 @@
+
+        # specific arguments must be passed in with the correct format
+        # (argspecs)- we don't need to do anything special here...
+
+        # create instance of ArgumentParser (pass along kwargs)
+        super(ArgParser, self).__init__(**kwargs)
+
+        cli_arguments = (compulsory_arguments + central_arguments +
+                         specific_arguments)
+
+        # automatically add all of the arguments
+        self.add_arguments(cli_arguments)
+        # Done. Now we can get the arguments with self.args()
+
+    @staticmethod
+    def _args_from_selected_central_args(arg_name_list):
+        """Return a list of argument specifications from the
+        ArgParser.CENTRALIZED_ARGUMENTS, given a list of keys (argument names).
+
+        Args:
+            arg_name_list (list):
+              A list containing the keys required from the centralized argument
+              dictionary.
+
+        Returns:
+            list:
+              The argument specifications which may later be used to add
+              arguments to the parser.
+        """
+        return [ArgParser.CENTRALIZED_ARGUMENTS[arg_name] for arg_name
+                in arg_name_list]
+
+    def add_arguments(self, argspec_list):
+        """Adds a list of arguments to the ArgumentParser.
+
+        The input argspec_list is a list of argument specifications, where each
+        element (argument specification) is a tuple/list of length 1 or 2.
+        The first element of an argument specification is a list of strings
+        which the name/flags used to add the argument.
+        (Optionally) the second element of the argument spec shall be a
+        dictionary containing the keyword arguments which are passed into the
+        add_arguments() method.
+
+        Args:
+            argspec_list (list):
+              A list containing the specifications required to add the
+              arguments (see above)
+
+        Raises:
+            AttributeError:
+                Notifies the user if any of the argument specifications has
+                the wrong length (not 1 or 2).
+        """
+        for argspec in argspec_list:
+            # each should be a list/tuple of length 1 or 2 (no more):
+            if len(argspec) == 1:
+                argflags, argkwargs = (argspec[0], {})
+            elif len(argspec) == 2:
+                argflags, argkwargs = argspec
+            else:
+                # AttributeError most appropriate?
+                # can't assign to argflags/argkwargs
+                raise AttributeError("The argument specification has an "
+                                     "unexpected length.")
+            self.add_argument(*argflags, **argkwargs)
+
+    def args(self):
+        """Returns the arguments passed in through the command line.
+
+        Automatically calls the parse_args method to return the arguments.
+        This means we do not necessarily need to call this manually in order
+        access the arguments.
```

Comment:
Could you add the comment as to why this function is necessary please.

Reply:
Will do. It's entirely stylistic anyway. I just liked the idea of having `args()` as an alias to `parse_args()`.

Reply:
I'm not sure if this actually makes things clearer,

Reply:
Caroline, you're right..

Generally in most of the CLIs, we do something along the lines of:
```
args = parser.parse_args()
function_using(args.my_cli_arg)
```
The idea behind using this function was so we could do `parser.args().my_cli_arg`, without actually having to call `parse_args()` multiple times (quick testing shows this doesn't seem to break anything though)  - I just thought it looked nicer to ask for the `args()`, since they're what are what you want (easier to remember), rather than remember that `args = parser.parse_args()`

(I'm not sure what the overhead on this is, but I suspect it's probably not worth caching the result in this way, but since it was only a couple of lines, I didn't see it as being a big problem.)

If there's heavy resistance against including this, I'm happy to remove it, but I really didn't think it would be a big deal?